effekt/jvm/src/main/scala/effekt/Server.scala

```diff
@@ -107,12 +107,22 @@ trait LSPServer extends kiama.util.Server[Tree, ModuleDecl, EffektConfig, Effekt
   } yield info
 
   // The implementation in kiama.Server does not support file sources
+  def toURI(filename: String): String = {
+    if (filename startsWith "file://")
+      filename
+    else
+      if (filename startsWith "./")
+        "file://" ++ Filenames.cwd() ++ "/" ++ Filenames.dropPrefix(filename, ".")
```

Comment:
I am not sure using `/` directly here is a good idea. It might be problematic with Windows users.

Reply:
Probably it would also be better to patch directly here: 

https://github.com/effekt-lang/kiama/blob/main/jvm/src/main/scala/kiama/util/Server.scala#L349-L363

Reply:
Hmm.. maybe not, looking at the code of override `locationOfNode` below I don't understand why we have the override.

Reply:
Regarding windows: it shouldn't be an issue if the implementation of the windows client adheres to the LSP:

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#uri

which sends it according to

https://tools.ietf.org/html/rfc3986

that is:
```
 foo://example.com:8042/over/there?name=ferret#nose
  \_/   \______________/\_________/ \_________/ \__/
   |           |            |            |        |
scheme     authority       path        query   fragment
   |   _____________________|__
  / \ /                        \
  urn:example:animal:ferret:nose
```

Reply:
Yes, I think `/` is used as path separator in URIs independent of any OS, so it should not be an issue here.

Wrt where to patch you certainly know better than me. But the implementation of `locationOfNode` in https://github.com/effekt-lang/kiama/blob/main/jvm/src/main/scala/kiama/util/Server.scala#L349-L363
does not deal with source filenames in nodes directly, hence the override. (At least, that's what I gathered and what the comment says.)